spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala

```diff
@@ -0,0 +1,134 @@
+package filodb.repair
+
+import java.io.File
+import java.time.Instant
+import java.time.format.DateTimeFormatter
+
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.StrictLogging
+import monix.execution.Scheduler
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+
+import filodb.cassandra.FiloSessionProvider
+import filodb.cassandra.columnstore.CassandraColumnStore
+import filodb.core.{DatasetRef, GlobalConfig}
+import filodb.core.store.ScanSplit
+
+/**
+  * Contains all the objects necessary for performing worker tasks. Is constructed from SparkConf
+  * and is cached.
+  */
+class ChunkCopier(conf: SparkConf) {
+  private def openConfig(str: String) = {
+    val sysConfig = GlobalConfig.systemConfig.getConfig("filodb")
+    ConfigFactory.parseFile(new File(conf.get(str))).getConfig("filodb").withFallback(sysConfig)
+  }
+
+  // Both "source" and "target" refer to file paths which define config files that have a
+  // top-level "filodb" section and a "cassandra" subsection.
+  val sourceConfig = openConfig("spark.filodb.chunkcopier.source.configFile")
+  val targetConfig = openConfig("spark.filodb.chunkcopier.target.configFile")
+
+  val sourceCassConfig = sourceConfig.getConfig("cassandra")
+  val targetCassConfig = targetConfig.getConfig("cassandra")
+
+  val sourceDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.source.dataset"))
+  val targetDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.target.dataset"))
+
+  // Examples: 2019-10-20T12:34:56Z  or  2019-10-20T12:34:56-08:00
+  private def parseDateTime(str: String) = Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(str))
```

Comment:
Do you mind making this configuration consistent with DownsamplerMain? Perhaps the following line can also use the above approach ?

```
    val userTimeInPeriod: Long = spark.sparkContext.getConf.get("spark.filodb.downsampler.userTimeOverride",
      s"${System.currentTimeMillis() - settings.downsampleChunkDuration}").toLong
```

Reply:
Which configuration consistent? You'd prefer that the timestamp be specified as seconds from 1970? The userTimeInPeriod line is capturing a duration, not an instant, and so it's different.

Reply:
I would like userTimeInPeriod to be specified as a string like in your PR

Reply:
But that's a duration, and it has a different syntax. Duration syntax is less standardized and many are unfamiliar with it.

Reply:
Will fix in a separate PR.